servlet/src/main/scala/org/http4s/servlet/ServletIo.scala

```diff
@@ -183,6 +192,55 @@ final case class NonBlockingServletIo[F[_]: Async](chunkSize: Int) extends Servl
       }
     }
 
+  /* The queue implementation is influenced by ideas in jetty4s
+   * https://github.com/IndiscriminateCoding/jetty4s/blob/0.0.10/server/src/main/scala/jetty4s/server/HttpResourceHandler.scala
+   */
+  override def requestBody(
+      servletRequest: HttpServletRequest,
+      dispatcher: Dispatcher[F],
+  ): Stream[F, Byte] = {
+    sealed trait Read
+    final case class Bytes(chunk: Chunk[Byte]) extends Read
+    case object End extends Read
+    final case class Error(t: Throwable) extends Read
```

Comment:
I ran some crude benchmarks on queues like this, and this wrapper doesn't make a difference.  jetty4s just has a queue of Any.  I like our typesafety here unless someone can show me in a profiler why it's bad.